lib/oauther.ex

```diff
@@ -94,7 +100,7 @@ defmodule OAuther do
   end
 
   defp nonce do
-    :crypto.rand_bytes(32) |> Base.encode64
+    :crypto.rand_bytes(16) |> Hexate.encode
```

Comment:
I've just tried to reach their API, it seem does work with previous nonce:
```
iex(7)> creds = OAuther.credentials(consumer_key: "L1cy4_dpdUNzcvY0692mUw", consumer_secret: "..", token: "RaWQwjjLw4ck0qVkQkC9VuebFtCmI9Ah", token_secret: "..")
%OAuther.Credentials{consumer_key: "L1cy4_dpdUNzcvY0692mUw",
 consumer_secret: "..", method: :hmac_sha1,
 token: "RaWQwjjLw4ck0qVkQkC9VuebFtCmI9Ah",
 token_secret: ".."}
iex(8)> params = OAuther.sign("get", url, [], creds)                                                                           [{"oauth_signature", "JMUpTbz9tbP1+Cq9E8oVnxbB/WU="},                               
 {"oauth_consumer_key", "L1cy4_dpdUNzcvY0692mUw"},
 {"oauth_nonce", "ikyMWGp2vB39gIgPlxC75NCxpJiTov38R17rsLBYU/o="},
 {"oauth_signature_method", "HMAC-SHA1"}, {"oauth_timestamp", 1442058099},
 {"oauth_version", "1.0"}, {"oauth_token", "RaWQwjjLw4ck0qVkQkC9VuebFtCmI9Ah"}]
iex(9)> {header, req_params} = OAuther.header(params)                                                                          {{"Authorization",                                                                  
  "OAuth oauth_signature=\"JMUpTbz9tbP1%2BCq9E8oVnxbB%2FWU%3D\", oauth_consumer_key=\"L1cy4_dpdUNzcvY0692mUw\", oauth_nonce=\"ikyMWGp2vB39gIgPlxC75NCxpJiTov38R17rsLBYU%2Fo%3D\", oauth_signature_method=\"HMAC-SHA1\", oauth_timestamp=\"1442058099\", oauth_version=\"1.0\", oauth_token=\"RaWQwjjLw4ck0qVkQkC9VuebFtCmI9Ah\""},
 []}
iex(10)> {:ok, _, _, ref} = :hackney.get(url, [header])                                                                        {:ok, 200,                                                                           
 [{"Date", "Sat, 12 Sep 2015 11:42:17 GMT"}, {"Server", "Apache"},
  {"X-Node", "10-70-154-62-eucentral1bprod, api_com"},
  {"Cache-Control", "private"},
  {"Set-Cookie",
   "bse=c7a6987f19222d9df3d9344b3180515a; Domain=.yelp.com; Path=/; HttpOnly"},
  {"Set-Cookie",
   "yuv=K3H8YarSlujUG6pt7wHFRlKr5r0yTbGW8KtmroNhz9FygpwOckZgEVk8UBdClPwEscCfHdTHPxgDMERRpv1ZFQaXehUZoc2t; Domain=.yelp.com; Max-Age=630720000; Path=/; expires=Fri, 07-Sep-2035 11:42:17 GMT"},
  {"Content-Length", "30934"}, {"Vary", "Accept-Encoding,User-Agent"},
  {"Content-Type", "application/json; charset=UTF-8"}, {"X-Mode", "ro"},
  {"X-Proxied", "10-70-2-86-eucentral1bprod"}], #Reference<0.0.0.1162>}
iex(11)> :hackney.body(ref)
{:ok,
 "{\"region\": {\"span\": {\"latitude_delta\": 0.047103329311269704, \"longitude_delta\": 0.047700768775001734}, \"center\": {\"latitude\": 37.7857050957676, \"longitude\": -122.416907667625}}, \"total\": 12649, \"businesses\": [{\"is_claimed\": false, \"rating\": 4.5, \"mobile_url\": \"http://m.yelp.com/biz/rove-kitchen-san-francisco\", \"rating_img_url\": \"http://s3-media2.fl.yelpcdn.com/assets/2/www/img/99493c12711e/ico/stars/v1/stars_4_half.png\", \"review_count\": 78, \"name\": \"Rove Kitchen\", \"snippet_image_url\": \"http://s3-media2.fl.yelpcdn.com/photo/lU_THuj3p7fls2BIc7d8aw/ms.jpg\", \"rating_img_url_small\": \"http://s3-media2.fl.yelpcdn.com/assets/2/www/img/a5221e66bc70/ico/stars/v1/stars_small_4_half.png\", \"url\": \"http://www.yelp.com/biz/rove-kitchen-san-francisco\", \"snippet_text\": \"Extraordinary food here, definitely the best burger in SF.\\n\\nI've been here a few times now, and it has yet to disappoint.  The prices are a bit high, but...\", \"image_url\": \"http://s3-media2.fl.yelpcdn.com/bphoto/cmxc2lnlpIFFvFIhyfHGEA/ms.jpg\", \"categories\": [[\"American (New)\", \"newamerican\"]], \"rating_img_url_large\": \"http://s3-media4.fl.yelpcdn.com/assets/2/www/img/9f83790ff7f6/ico/stars/v1/stars_large_4_half.png\", \"id\": \"rove-kitchen-san-francisco\", \"is_closed\": false, \"location\": {\"cross_streets\": \"Jones St \\u0026 Ophir Aly\", \"city\": \"San Francisco\", \"display_address\": [\"678 Post St\", \"Lower Nob Hill\", \"San Francisco, CA 94109\"], \"geo_accuracy\": 9.5, \"neighborhoods\": [\"Lower Nob Hill\"], \"postal_code\": \"94109\", \"country_code\": \"US\", \"address\": [\"678 Post St\"], \"coordinate\": {\"latitude\": 37.7877664572471, \"longitude\": -122.412939658872}, \"state_code\": \"CA\"}}, ..}
```
There is branch that I've used https://github.com/lexmag/oauther/tree/extract-query-params for test.
Same changes to your patch, but we need to keep using `String.downcase(host)`, and use `URI.query_decoder/1` to remove intermediate map and what's more important is to keep duplicates in the query params.
Could you confirm that it works for your case?
If so, please apply changes in it and we ready to merge. :bowtie: